### PR TITLE
docs: sonarqube configure docs

### PR DESCRIPTION
**Pull request by abeizn**
### ⚠️ &nbsp;&nbsp;Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have `npm run build` and `npm run serve` locally before submitting this PR
- [x] I have read through the [Contributing](https://devlake.apache.org/community/) Documentation

# Summary

sonarqube configure docs

### Does this close any open issues?
related to https://github.com/apache/incubator-devlake/issues/4462

### Screenshots
image
image
image
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: docs/Configuration/AdvancedMode.md [hunk 1]
```diff
@@ -325,6 +326,27 @@ Below is an example for collecting a bitbucket repo.
 - `owner`: the owner of the repository.
 - `repo`: the bitbucket repository name.
 
+### 10. Sonarqube
```
Comment (Startrekzky) [inline docs/Configuration/AdvancedMode.md]: If sonarqube does not appear in the plugin name or URL, use SonarQube.

File: docs/Configuration/AdvancedMode.md [hunk 2]
```diff
@@ -325,6 +326,27 @@ Below is an example for collecting a bitbucket repo.
...
+        "connectionId": 1,
+        "projectKey": "testDevLake"
+      }
+    }
+  ]
+]
+```
+
+- `connectionId`: The ID of your sonarqube connection at page http://localhost:4000/connections/sonarqube.
+- `projectKey`: the project name of the sonarqube.
```
Comment (Startrekzky) [inline docs/Configuration/AdvancedMode.md]: ProjectKey or ProjectName? If it is ProjectKey, please show users how to get it with a URL to the SonarQube doc or a screenshot.

File: docs/Configuration/Sonarqube.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+This should be a valid REST API endpoint
+
+- `https://YOUR_DOMAIN:YOUR_PORT/api/`
+
+The endpoint url should end with `/`.
+
+#### Auth Token(s)
+
+Sonarqube personal access tokens are required to add a connection. Learn about [how to create a Sonarqube personal access token](https://sonarqube.inria.fr/sonarqube/documentation/user-guide/user-token/).
```
Comment (Startrekzky) [inline docs/Configuration/Sonarqube.md]: 1. SonarQube personal access tokens
2. Are there requirements for token permissions?

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 4]
```diff
@@ -496,30 +496,110 @@ A cicd_task is a single job of ci/cd.
...
+#### cq_projects
+
+| **field**            | **type** | **length** | **description**                                                                     | **key** |
+| :------------------- | :------- | :--------- | :---------------------------------------------------------------------------------- | :------ |
+| `id`                 | varchar  | 255        | This key is generated based on details from the original plugin                     | PK      |
+| `name`               | varchar  | 255        | The name of the project in SonarQube                                                |         |
+| `qualifier`          | varchar  | 255        | The type of project. Examples include "TRK" for regular projects and "VW" for views |         |
+| `visibility`         | varchar  | 64         | The visibility of the project. Examples include "public" and "private"              |         |
+| `last_analysis_date` | datatime | 3          | The date and time of the most recent analysis of the project                        |         |
+| `commit_sha`         | varchar  | 128        | The SHA1 hash of the latest commit associated with the project                      |         |
```
Comment (Startrekzky) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: "The SHA1 hash of the latest commit associated with the project", is this true?

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 5]
```diff
@@ -496,30 +496,110 @@ A cicd_task is a single job of ci/cd.
 | `finished_date` | datetime        | 3          | when did this task finish                                       |         |
 | `environment`   | varchar         | 255        | To indicate the environment in which the task is running        |         |
 
+### Domain 5 - Code Quality
```
Comment (Startrekzky) [inline docs/DataModels/DevLakeDomainLayerSchema.md:499]: We can add a line here: The names of tables in the 'Code Quality' domain will start with a prefix `cq_`

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 6]
```diff
@@ -496,30 +496,110 @@ A cicd_task is a single job of ci/cd.
...
+| `project_key`               | varchar  | 255        | The key of the project that the issue belongs to                        |         |
+| `line`                      | bigint   |            | The line number where the issue was found                               |         |
+| `status`                    | varchar  | 255        | The status of the issue                                                 |         |
+| `message`                   | longtext |            | The message associated with the issue                                   |         |
+| `debt`                      | bigint   |            | The estimated time required to fix the issue                            |         |
+| `effort`                    | bigint   |            | The effort required to fix the issue                                    |         |
+| `commit_author_email`       | varchar  | 255        | The email address of the author of the commit that introduced the issue |         |
+| `assigne`                   | varchar  | 255        | The person assigned to fix the issue                                    |         |
```
Comment (Startrekzky) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: typo: assignee

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 7]
```diff
@@ -496,30 +496,110 @@ A cicd_task is a single job of ci/cd.
...
+| `tags`                      | varchar  | 255        | Any tags associated with the issue                                      |         |
+| `type`                      | varchar  | 255        | The type of the issue                                                   |         |
+| `scope`                     | varchar  | 128        | The scope of the issue                                                  |         |
+| `start_line`                | bigint   | 255        | The starting line of the issue                                          |         |
+| `end_line`                  | bigint   | 255        | The ending line of the issue                                            |         |
+| `start_offset`              | bigint   | 255        | The starting offset of the issue                                        |         |
+| `end_offset`                | bigint   | 255        | The ending offset of the issue                                          |         |
+| `vulnerability_probability` | varchar  | 100        | The probability of the issue being a vulnerability                      |         |
```
Comment (Startrekzky) [inline docs/DataModels/DevLakeDomainLayerSchema.md:538]: Where are `vulnerability_probability` and `security_category` from? Issue or hotspot API?

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 8]
```diff
@@ -496,30 +496,110 @@ A cicd_task is a single job of ci/cd.
...
+| `security_category`         | varchar  | 100        | The security category of the issue                                      |         |
+| `creation_date`             | datetime | 3          | The date and time when the issue was created                            |         |
```
Comment (Startrekzky) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: Shall we use `created_date` and `updated_date` for consistency?

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 9]
```diff
@@ -496,30 +496,110 @@ A cicd_task is a single job of ci/cd.
...
+#### cq_issue_code_blocks
+
+| **field**      | **type** | **length** | **description**                                                                       | **key** |
+| :------------- | :------- | :--------- | :------------------------------------------------------------------------------------ | :------ |
+| `id`           | varchar  | 255        | This key is generated based on details from the original plugin                       | PK      |
+| `issue_key`    | varchar  | 255        | A string that stores the key of the issue that the code block is associated with      |         |
+| `component`    | varchar  | 255        | A string that stores the name of the component that the code block is associated with |         |
+| `start_line`   | bigint   | 255        | An integer that stores the line number where the code block starts                    |         |
+| `end_line`     | bigint   | 255        | An integer that stores the line number where the code block ends                      |         |
+| `start_offset` | bigint   | 255        | An integer that stores the line number where the code block ends                      |         |
```
Comment (Startrekzky) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: Should it be "An integer that stores the offset where the code block starts"?

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 10]
```diff
@@ -496,30 +496,110 @@ A cicd_task is a single job of ci/cd.
...
+| `end_offset`   | bigint   | 255        | An integer that stores the offset where the code block ends                           |         |
+| `msg`          | longtext |            | A long text field that stores the message associated with the code block              |         |
+
+#### cq_file_metrics
```
Comment (Startrekzky) [inline docs/DataModels/DevLakeDomainLayerSchema.md:556]: @warren830 added a few columns to this table.

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 11]
```diff
@@ -496,30 +496,110 @@ A cicd_task is a single job of ci/cd.
...
+
+| **field**                    | **type** | **length** | **description**                                                  | **key** |
+| :--------------------------- | :------- | :--------- | :--------------------------------------------------------------- | :------ |
+| `id`                         | varchar  | 255        | This key is generated based on details from the original plugin  | PK      |
+| `project_key`                | varchar  | 255        | The key of the project that the issue belongs to                 | PK      |
+| `file_name`                  | longtext |            | longtext fields that store the name of the file                  |         |
+| `file_path`                  | longtext |            | longtext fields that store the path of the file                  |         |
+| `file_language`              | longtext |            | longtext fields that store the language of the file              |         |
+| `code_smells`                | bigint   |            | numeric fields that store various metrics of the file            |         |
```
Comment (Startrekzky) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: 1. The description can be updated to: Code smells of this file.
2. Same goes to the description below

File: docs/Overview/SupportedDataSources.md [hunk 12]
```diff
@@ -9,77 +9,80 @@ sidebar_position: 5
...
+| PagerDuty (WIP)  | Issue Tracking                                                              | Cloud                                | Not Available           | `pagerduty`                 | Full Refresh                                                   |
+| Feishu (WIP)     | Calendar                                                                    | Cloud                                | Not Available           | `feishu`                    | Full Refresh                                                   |
+| AE               | Source Code Management                                                      | On-prem                              | Not Available           | `ae`                        | Full Refresh                                                   |
+| Sonarqube        | CODE QUALITY                                                                | Community Edition v8.2+              | Available               | `sonarqube`                 | Full Refresh                                                   |
```
Comment (Startrekzky) [inline docs/Overview/SupportedDataSources.md]: Let's use SonarQube(Server) if we're not sure if SonarQube cloud is supported.

File: docs/Plugins/sonarqube.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+Check out the [Sonarqube entities](/Overview/SupportedDataSources.md#data-collection-scope-by-each-plugin) collected by this plugin.
+
+## Data Refresh Policy
+
+Check out the [data refresh policy](/Overview/SupportedDataSources.md#sonarqube) of this plugin.
+
+## Metrics
+
+Metrics that can be calculated based on the data collected from Sonarqube:
```
Comment (Startrekzky) [inline docs/Plugins/sonarqube.md]: Update with: Most of SonarQube metrics are collected and can be found in DevLake's SonarQube dashboard.

**Review by Startrekzky — CHANGES_REQUESTED** [review]

File: docs/Configuration/Sonarqube.md (hunk 3 above)
Comment (abeizn) [reply docs/Configuration/Sonarqube.md]: I created a token, but I really didn't find a place with permission control

File: docs/Configuration/AdvancedMode.md [hunk 14]
```diff
@@ -325,6 +326,32 @@ Below is an example for collecting a bitbucket repo.
...
+        "connectionId": 1,
+        "projectKey": "testDevLake"
+      }
+    }
+  ]
+]
+```
+
+- `connectionId`: The ID of your SonarQube connection at page http://localhost:4000/connections/sonarqube.
+- `projectKey`: the project name of the SonarQube. To find the project key in SonarQube, you can follow these steps:
```
Comment (Startrekzky) [inline docs/Configuration/AdvancedMode.md]: The project key of the SonarQube. To find the project key in SonarQube, please follow the steps:

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 15]
```diff
@@ -612,29 +700,29 @@ metrics, such as _'No. of Issue closed by contributor', 'No. of commits by contr
...
-| ------------- | -------- | ---------- | ----------------------------- | ------- |
-| `name`        | varchar  | 255        | name for project              | PK      |
-| `description` | longtext |            | description of the project    |         |
-| `created_at`  | datetime | 3          | created time of project       |         |
-| `updated_at`  | datetime | 3          | last updated time of project  |         | 
+| **field**     | **type** | **length** | **description**              | **key** |
+| ------------- | -------- | ---------- | ---------------------------- | ------- |
+| `name`        | varchar  | 255        | name for project             | PK      |
+| `description` | longtext |            | description of the project   |         |
+| `created_at`  | datetime | 3          | created time of project      |         |
```
Comment (Startrekzky) [inline docs/DataModels/DevLakeDomainLayerSchema.md:707]: Shall we use `created_date` and `updated_date`?

**Review by Startrekzky — CHANGES_REQUESTED** [review]

**Review by Startrekzky — APPROVED** [review]
LGTM

**Pull request merged**